api/v1alpha1/types.go

```diff
@@ -0,0 +1,118 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+type ProvisionerID string
+
+// BundleClassSpec defines the desired state of BundleClass
+type BundleClassSpec struct {
+	Provisioner ProvisionerID `json:"provisioner"`
+
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:resource:categories=rukpak,scope=Cluster
+// +kubebuilder:subresource:status
+
+// BundleClass is the Schema for the provisionerclasses API
+type BundleClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec BundleClassSpec `json:"spec"`
+}
+
+// +kubebuilder:object:root=true
+
+// BundleClassList contains a list of BundleClass
+type BundleClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BundleClass `json:"items"`
+}
+
+// Source locates content to stage.
+// TODO(njhale): consider runtime.RawExtension as a way to let provisioners define their own schemes for sourcing content;
+// e.g. inline kubernetes manifests, resource references, etc.
```

Comment:
That's an interesting thought - (nit) can we deflate this TODO so it doesn't also appear when rendering manifests. Open an RFE issue?

Reply:
Sure -- I could also figure out how to use [git-notes](https://git-scm.com/docs/git-notes).